### ISSUE TITLE
Degrade, Upgrade Stations for portal gun.

### ISSUE THREAD

**Issue by Nikita74RU** [BEEmod/BEE2-items#2662]
A pedestal, portal gun stands on it, if you pick the portal gun standing on it, it gives the portal gun you had to the pedestal, resulting the pedestal to shut off. and lower down. Seen in Portal Mod: Factum Solus.
If you choose Weighted Cube, it will upgrade gun to Duel, if you have single gun. And same with orange only gun. You can choose which colour to get away, Weighted- Upgrade to duel, Cube button- Grab away blue portal Ball Receptacle- Grab away orange portal.

**Comment by HugoBDesigner** [comment]
I personally think this should be the default behavior if the map has you grab a gun while already owning that portal gun color. So for instance, if you have both portals and grab a dual portal device, you get them both removed.

**Comment by Nikita74Russian** [comment]
Also if you have a orange colour it will upgrade to blue if you picked the blue station. If you have orange and you pick the orange station, you lose orange portal gun. Same with others.

**Comment by Nikita74Russian** [comment]
Orange colour in the gun= Dual portal device Orange colour in the gun= press onto station= no portal gun

**Comment by HugoBDesigner** [comment]
Yes, that's literally what I just explained. Thanks for needless repetition
